app/proc/telegram.go

```diff
@@ -100,11 +114,21 @@ func (client TelegramClient) sendAudio(channelID string, item feed.Item) (*tb.Me
 		return nil, err
 	}
 	defer os.Remove(tmpFile.Name())
-	if _, err := io.Copy(tmpFile, httpBody); err != nil {
+
+	if _, err = io.Copy(tmpFile, httpBody); err != nil {
 		return nil, err
 	}
-	if err := tmpFile.Close(); err != nil {
-		return nil, err
+	if closeErr := tmpFile.Close(); closeErr != nil {
+		return nil, closeErr
+	}
+
+	var dur int
+	if item.Duration != "" { // item may have duration published, if not, try to get it from mp3 file
```

Comment:
I propose getting the file duration first and then falling back to the original item duration. Getting the duration from the file is reasonably precise and reliable, and it's safe to have it as a default method.

Reply:
the only case with the filled item.Duration is the item coming from yt processing. Only those suckers have duration published. This is done with exactly the same duration service, so " getting the file duration first" will be just a duplicated call